### DISCUSSION TITLE
module installation error?

### DISCUSSION THREAD

**Discussion by cageman1970** [AllskyTeam/allsky-modules#74] [Q&A]
I started with a complete new install (latest version of Bookworm and the latest repository from allsky.  Tried to install the modules as instructed on https://github.com/AllskyTeam/allsky-modules. That screwed my installation up completely.   The module screen in Allsky was totally empty, seemed that all modules were deleted. 

I followed the installation guide. after the git clone and the./install.sh command (Not understanding at all what was meant by 'if you are using any other branch other than master branch...'). it resulted in an extra folder in the home/pi directory  with all the modules. Running the ./install.sh and the module installer menu was not intuitive; i could select the modules i wanted but with the right/left arrow keys i could not end up at the OK or Cancel buttons. So hoped that with an 'Enter' everything should work out. that didnt, all the modules including the original ones were gone. 

So i had to do it all over again, clean install, etc. Ended up trying to download the modules i wanted (dewheater and openweathermap) and copy them to allsky (just the .py files). It looks to work, they show up in de module manager but the needed files like openweather.json and allskydew.json are not created  in allsky/tmp/extra. the bug report delivers nothing (allsky_dewheater.py and allsky_openweathermap.py have a run time from 0.00 seconds).

I dont want to do a clean install for the third time, so am hesistant to try to follow the instructions on the module github page again. Are these instructions valid?

**Comment by Alex-developer** [comment]
Hi
I suspect this has been caused by installing the allsky_test module, this is not intended to be used and in the docs it mentions not using it - I will remove it as it was my mistake to include it.

The installer does a lot more than just copy the .py files so you cannot simply copy the files into the allsky directory, in fact this is the wrong place for the extra modules. 

The debug option is only intended to display the results of a module running, it will not display fatal errors such as missing libraries. These  errors will be in /var/log/allsky and /var/log/allskyperiodic (Depending upon the flow the module is in)

The instructions are valid and I have installed the modules hundreds of times whilst testing, I will check the keys used to navigate around the installer when I have a pi to hand. The navigation is supposed to be the same as the raspi-config utility but I may have got something wrong, 

Without knowing exactly what has been copied where its hard to give you a 100% fix but I would try

1) Remove any files that you have manually copied into allsky
2) Delete all files and folders from inside the /opt/allsky/modules/ folder
3) Check the module manager to make sure the core modules are visible, if they are not then its likely there is a rogue file still in the allsky modules directory
4) clone the module installer
5) run ./install.sh
6) Select the install option
7) Use the arrow keys to navigate up and down the modules list and the <space> bar to select or deselect a module - DO NOT install the test module
8) When happy hit the <tab> key and you will then be able to select the <ok> button
9) Let the installer do its stuff

Hope that helps but feel free to shout if you are still having an issue

Cheers

Alex
